### PR TITLE
dont use uuids for bossbars, use userconnection

### PR DESCRIPTION
**Pull request by creeper123123321**
may fix ViaVersion/VIaFabric#27
not tested.

### CODE REVIEW

**Comment by creeper123123321** [comment]
Works.

File: common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java [hunk 1]
```diff
@@ -202,6 +229,18 @@ private void sendPacket(UUID uuid, PacketWrapper wrapper) {
         }
     }
 
+    private void sendPacketConnection(UserConnection conn, PacketWrapper wrapper) {
+        if (conn.getProtocolInfo() == null || conn.getProtocolInfo().getProtocolVersion() < ProtocolVersion.v1_9.getId()) {
+            connections.remove(conn);
+            return;
+        }
+        try {
+            wrapper.send(Protocol1_9To1_8.class);
```
Comment (creeper123123321) [inline common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java:238]: will show an error if this is using viabackwards

File: common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java [hunk 2]
```diff
@@ -91,6 +89,17 @@ public BossBar addPlayer(UUID player) {
         return this;
     }
 
+    @Override
+    public BossBar addConnection(UserConnection conn) {
+        if (!connections.contains(conn)) {
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java:86]: Directly add and check the return value

File: common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java [hunk 3]
```diff
@@ -101,6 +110,15 @@ public BossBar removePlayer(UUID uuid) {
         return this;
     }
 
+    @Override
+    public BossBar removeConnection(UserConnection conn) {
+        if (connections.contains(conn)) {
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/boss/CommonBoss.java:102]: Directly remove and check return value

File: common/src/main/java/us/myles/ViaVersion/api/boss/BossBar.java [hunk 4]
```diff
@@ -112,13 +121,21 @@ public BossBar removePlayer(T player) {
...
      */
     public abstract BossBar removePlayer(UUID uuid);
 
+    /**
+     * Removes the bossbar from a player connection.
+     *
+     * @param conn The UserConnection
+     * @return The BossBar object
+     */
+    public abstract BossBar removeConnection(UserConnection conn);
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/api/boss/BossBar.java:137]: Probably should remove the uuid mappings

Comment (KennyTV) [reply common/src/main/java/us/myles/ViaVersion/api/boss/BossBar.java:137]: Ah fixed while I was reviewing :p

File: common/src/main/java/us/myles/ViaVersion/api/boss/BossBar.java [hunk 5]
```diff
@@ -80,13 +81,21 @@ public BossBar addPlayer(T player) {
     }
 
     /**
-     * Show the bossbar to a player (uuid)
+     * Show the bossbar to a player (uuid).
```
Comment (KennyTV) [inline common/src/main/java/us/myles/ViaVersion/api/boss/BossBar.java:84]: Could maybe add a line on why there's adders for both uuid and connection (and why direct connection is preffered)

**Comment by FormallyMyles** [comment]
:shipit:

**Pull request merged**